packages/cosmic-swingset/lib/ag-solo/web.js

```diff
@@ -101,24 +108,21 @@ export function makeHTTPListener(basedir, port, host, rawInboundCommand) {
     return true;
   };
 
-  // accept messages on some well-known endpoints
-  // todo: later allow arbitrary endpoints?
-  for (const ep of ['/vat', '/wallet-public', '/api']) {
-    app.post(ep, (req, res) => {
-      if (ep !== '/api' && !validateOrigin(req)) {
-        res.json({ ok: false, rej: 'Invalid Origin' });
-        return;
-      }
+  // accept POST messages to arbitrary endpoints
+  app.post('*', (req, res) => {
+    if (!validateOrigin(req)) {
```

Comment:
as before, I want to consider origin checks to be weak, and build a proper auth mechanism soon

Reply:
Absolutely!